packages/extension/src/background/transactions/transactionExecution.ts

```diff
@@ -29,15 +29,45 @@ export const executeTransactionAction = async (
     )
   }
 
-  if (account !== undefined) {
+
+  addTransaction({
+    account: account,
+    hash: transaction.result.txId,
+    meta: {
+      request: transaction,
+    },
+  })
+
+  return { signature: finalSignature }
+}
+
+export const executeTransactionsAction = async (
+  transactions: ReviewTransactionResult[],
+  { wallet }: BackgroundService,
+  networkId: string,
+): Promise<{ signatures: string[] }> => {
+  const signatures: string[] = []
+  for (const transaction of transactions) {
+    const account = await wallet.getAccount({
+      address: transaction.params.signerAddress,
+      networkId: networkId,
+    })
+
+    const { signature } = await wallet.signAndSubmitUnsignedTx(account, {
+      signerAddress: account.address,
+      unsignedTx: transaction.result.unsignedTx,
+    })
+
```

Comment:
Why is the signing code logic different from executeTransactionAction?

Reply:
Great question. I have tested that the `signatureOpt` in `executeTransactionAction` is not needed. I will remove it so it is consistent with `executeTransactionsAction`.